src/server/acl/acl_family.cc

```diff
@@ -468,6 +469,36 @@ void AclFamily::GetUser(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendSimpleString(acl);
 }
 
+void AclFamily::GenPass(CmdArgList args, ConnectionContext* cntx) {
+  if (args.length() > 1) {
+    (*cntx)->SendError(
+        "Unknown subcommand or wrong number of arguments for 'GENPASS'. Try ACL HELP.");
+    return;
+  }
+  uint32_t random_bits = 256;
+  if (args.length() == 1) {
+    auto requested_bits = facade::ArgS(args, 0);
+
+    if (!absl::SimpleAtoi(requested_bits, &random_bits) || random_bits == 0 || random_bits > 4096) {
+      return (*cntx)->SendError(
+          "ACL GENPASS argument must be the number of bits for the output password, a positive "
+          "number up to 4096");
+    }
+  }
+  std::random_device urandom("/dev/urandom");
+  std::independent_bits_engine<std::default_random_engine, CHAR_BIT, unsigned char>
+      random_bits_engine(urandom());
+  std::vector<unsigned char> data((random_bits + 3) / 4);
+  std::generate(begin(data), end(data), std::ref(random_bits_engine));
+
+  std::string response;
+  for (auto c : data) {
+    absl::StrAppend(&response, absl::Hex(c & 0x0F, absl::kNoPad));
+  }
+
```

Comment:
```
    std::random_device urandom("/dev/urandom");
    const size_t result_length = (random_bits + 3) / 4;
    constexpr size_t step_size = sizeof(decltype(urandom.max()));
    std::string response;
    for(size_t bytes_written = 0; bytes_written < result_length; bytes_written += step_size) {
      absl::StrAppend(&response, absl::Hex(urandom(), absl::kZeroPad8));
    }
  
    if(response.size() > result_length) {
      const size_t stride = response.size() - result_length;
      response.erase(response.end() - stride, response.end());
    }


```

Reply:
This will solve the problem :)

Reply:
Alright, but in the end I did not still understand why the independent_bits engine solution was no good? Is it because it calls the generator too many times?

Reply:
and also:

```
template<
    class Engine,
    std::size_t W,
    class UIntType
    class independent_bits_engine;
```

W is a compile time parameter but we know this number at runtime